app/models/katello/root_repository.rb

```diff
@@ -306,6 +307,16 @@ def ensure_valid_deb_constraints
       end
     end
 
+    def ensure_valid_retain_package_versions_count
+      return unless self.retain_package_versions_count
+      unless yum?
+        errors.add(:retain_package_versions_count, N_("is only allowed for Yum repositories."))
```

Comment:
When should I see this error pop up? I'm able to successfully do something like this: 

`hammer repository create --organization-id 1 --product-id 1 --content-type file --name "file 3" --retain-package-versions-count 1`

Reply:
Do you also see the retain package version being passed to katello and being set on the repo? I remember seeing a BZ around hammer ignoring that param.

Reply:
it gets passed to katello, but it isn't set on the repo because of this conditional: https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/repositories_controller.rb#L571.

But it looks like updating the repository will trigger the error message.

Reply:
> In the new repository screen, if I put a negative value and click "save" I get the error message. _However_, when I replace it with a positive value and click "save" nothing happens and the repository does not create. I don't see errors in the console. I have to refresh the page to save it with a valid value.
> 
> Are you seeing that too?

I see this..But I see this in master too with any validation error on the repo page, ex: duplicate name.

Reply:
okay, not a problem here then :)